### PR TITLE
Macro sizes API

### PR DESCRIPTION
**Pull request by edwardcwang**
This is the Hammer side only. Floorplanning DSL side forthcoming.

### CODE REVIEW

File: src/hammer-tech/hammer_tech.py [hunk 1]
```diff
@@ -384,6 +413,84 @@ def tech_defined_libraries(self) -> List[Library]:
...
+
+    def get_tech_macro_sizes(self) -> List[MacroSize]:
+        """
+        Compile a list of all macros which have size information, using LEF files.
+        This also considers any extra IP libraries.
+        :return: List of all macros' size information.
+        """
+        extra_libraries = self.get_extra_libraries()  # type: List[ExtraLibrary]
+
+        # Enhance the LEF file to also extract the name of the library
```
Comment (colinschmidt) [inline src/hammer-tech/hammer_tech.py]: I don't understand this comment. It doesn't appear to be about the function defined below it.

Comment (edwardcwang) [reply src/hammer-tech/hammer_tech.py]: Oops, there is a typo. Will fix.

**Comment by edwardcwang** [comment]
Here's the current plan:

- Nothing changes in the placement constraint API
- Floorplanning DSL gains the ability to read off all macros' sizes and can use it in floorplanning
- Floorplanning DSL emits hardmacro sizes

**Comment by colinschmidt** [comment]
Any reason not to merge?

**Comment by edwardcwang** [comment]
Apologies for the delay. Had to fix a minor merge conflict. Will merge ASAP, after checks finish running. If any other issues come up I'll just fix it in a future PR.

**Pull request merged**